src/app/Main/Body/QueryPage/Results/Content-Results.component.tsx

```diff
@@ -1,126 +1,125 @@
+                        _.includes(uniqIsolateRowValues, rowElement)
+                    ) {
+                        isolateCountProp.groups.forEach((isolateGroup) => {
+                            if (
+                                (isolateGroup[rowAttribute] === rowElement ||
+                                    isolateGroup[rowAttribute] === undefined) &&
+                                (isolateGroup[colAttribute] === colElement ||
+                                    isolateGroup[colAttribute] === undefined)
+                            ) {
+                                const statTableKey =
+                                    isolateGroup[colAttribute] === undefined
+                                        ? t("Results.TableHead")
+                                        : isolateGroup[colAttribute];
+                                tempStatTable[
+                                    statTableKey
+                                ] = (isolateGroup.count as unknown) as string;
+                            }
+                        });
+                    } else {
+                        tempStatTable[colElement] = "0";
+                    }
+                });
+                statisticTableDataAbs.push(tempStatTable);
             });
-        } else if (colAttribute.length === 0 && rowAttribute.length !== 0) {
-            const rowsWithIsolates = generateRowsWithIsolates(
-                dataToCountIsolates,
-                colAttribute,
-                rowAttribute,
-                rowValues,
-                [t("Results.TableHead")],
-                "row"
+
+            const nrOfSelectedIsolates = isolateCountProp.totalNumberOfIsolates;
+            const rowsWithPercent = calculateRowsWithPercent(
+                statisticTableDataAbs,
+                nrOfSelectedIsolates
             );
-            const rowsWithPercent = calculateRowsWithPercent(rowsWithIsolates, nrOfSelectedIsolates)
-            setState({
-                isCol: false,
-                isRow: true,
-            });
-            setColumnAttributes([t("Results.TableHead")]);
+            setColumnAttributes(colValues);
             setTable({
                 ...table,
-                statisticDataAbsolute: rowsWithIsolates,
+                statisticDataAbsolute: statisticTableDataAbs,
                 statisticDataPercent: rowsWithPercent,
             });
-        } else if (rowAttribute.length === 0 && colAttribute.length === 0) {
-            setState({
-                isCol: false,
-                isRow: false,
-            });
         }
     };
 
+    const ISOLATE_COUNT_URL: string = isolateCountURL + history.location.search;
+
+    const fetchIsolateCounted = async (): Promise<void> => {
+        const isolateCountResponse: Response = await fetch(ISOLATE_COUNT_URL);
+        const isolateCountProp: IsolateCountedDTO = await isolateCountResponse.json();
+
+        getTableContext(isolateCountProp);
+    };
+

```

Comment:
Consider adding an api service that does the fetching and mapping of the data. So that you have only one point in the code where the api fetching stuff happens. Also you don't want handle the DTO conversion stuff in your components but in an encapsulated part of the code. Every time you import some DTO model into a component it is kind of a code smell for me.
So to make it clear: The api service should be the layer that separates the api calls and dto handling from your components and internal data structures.

Reply:
This will change with the refactoring of the container-view-structure.